invenio_mail/tasks.py

```diff
@@ -79,4 +85,40 @@ def _send_email_with_attachments(data, attachments=[]):
             disposition = attachment.get("disposition")
         msg.attach(content_type=content_type, data=rawdata, disposition=disposition)
 
-    current_app.extensions["mail"].send(msg)
+    _send(msg, self)
+
+
+def _send(msg, task):
+    """Send emails and add exception handling."""
+    try:
+        current_app.extensions["mail"].send(msg)
+    except (
+        smtplib.SMTPRecipientsRefused,
+        smtplib.SMTPHeloError,
+        smtplib.SMTPSenderRefused,
+        smtplib.SMTPDataError,
+        smtplib.SMTPNotSupportedError,
+    ):
+        if task.request.retries <= current_app.config["MAIL_MAX_RETRIES"]:
+            raise task.retry(
+                countdown=int(random.uniform(2, 4) ** task.request.retries)
+            )
+        else:
+            current_app.logger.error("Mail could not be dispatched!")
+            if current_app.config["MAIL_LOG_FAILED_MESSAGES"]:
+                current_app.logger.info(msg.body)
+                if msg.attachments:
+                    for attachment in msg.attachments:
+                        current_app.logger.info(attachment.data)
```

Comment:
Is this needed in production? Can it be simply replaced by `logger.debug`?
And is it OK to log the entire attachment.data? Does it correspond to the content of the attachment? what is it a binary file?

Reply:
We don't need all the debug log messages, that's too much noise. in production we want the contents of mail and attachment (which is base64 encoded) to send it by hand.